test/e2e-shared-tests/mcd_config_drift.go

```diff
@@ -218,6 +231,62 @@ func (c configDriftTest) Run(t *testing.T) {
 				})
 			},
 		},
+		{
+			name:           "update degraded reason on drift",
+			rebootExpected: false,
+			testFunc: func(t *testing.T) {
+				mutateFileOnNode(t, c.ClientSet, c.node, configDriftFilename, "not-the-data")
+				assertNodeAndMCPIsDegraded(t, c.ClientSet, c.node, c.mcp, configDriftFilename)
+
+				node, err := c.ClientSet.CoreV1Interface.Nodes().Get(context.TODO(), c.node.Name, metav1.GetOptions{})
+				require.Nil(t, err)
+
+				firstReason := node.Annotations[constants.MachineConfigDaemonReasonAnnotationKey]
+
+				mutateFileOnNode(t, c.ClientSet, c.node, configDriftFilename, configDriftFileContents)
+				assertNodeAndMCPIsRecovered(t, c.ClientSet, c.node, c.mcp)
+
+				mutateFileOnNode(t, c.ClientSet, c.node, configDriftFilenameTwo, "incorrect data 2")
+				assertNodeAndMCPIsDegraded(t, c.ClientSet, c.node, c.mcp, configDriftFilenameTwo)
+
+				mutateFileOnNode(t, c.ClientSet, c.node, configDriftFilename, "not-the-data")
+				assertNodeAndMCPIsDegraded(t, c.ClientSet, c.node, c.mcp, configDriftFilename)
+
+				node, err = c.ClientSet.CoreV1Interface.Nodes().Get(context.TODO(), c.node.Name, metav1.GetOptions{})
+				require.Nil(t, err)
+
+				assert.Equal(t, node.Annotations[constants.MachineConfigDaemonReasonAnnotationKey], firstReason)
+
+				assertPoolReachesState(t, c.ClientSet, c.mcp, func(m mcfgv1.MachineConfigPool) bool {
+					trueConditions := []mcfgv1.MachineConfigPoolConditionType{
+						mcfgv1.MachineConfigPoolDegraded,
+						mcfgv1.MachineConfigPoolNodeDegraded,
+						mcfgv1.MachineConfigPoolUpdating,
+					}
+
+					falseConditions := []mcfgv1.MachineConfigPoolConditionType{
+						mcfgv1.MachineConfigPoolRenderDegraded,
+						mcfgv1.MachineConfigPoolUpdated,
+					}
+
+					return m.Status.DegradedMachineCount == 1 &&
+						allMCPConditionsTrue(trueConditions, m) &&
+						allMCPConditionsFalse(falseConditions, m)
+				})
+
+				mcp, err := c.ClientSet.MachineConfigPools().Get(context.TODO(), c.mcp.Name, metav1.GetOptions{})
+				require.Nil(t, err)
+				for _, condition := range mcp.Status.Conditions {
+					if condition.Type == mcfgv1.MachineConfigPoolNodeDegraded {
+						assert.True(t, strings.Contains(condition.Message, configDriftFilename))
```

Comment:
Optional: There is an `assert.Contains()` method which is a shortcut for comparison. I'd also suggest combining this with at status check as well:

```golang
assert.Equal(t, condition.Status, corev1.ConditionTrue)
```

Also optional: Each condition only occurs once in the `mcp.Status.Conditions` array. Once you've found the matching condition and performed your assertions against it, you can break out of the loop by using the `break` keyword. It's not required in this case because the number of elements in that array is pretty small, but it's something to get into a good habit of.